Fix known issues

AsiganTheSunk opened the pull request:
# Patch Notes for Uxio0
- Removed install-travis.sh
- Added installation lines in travis.yml (npm install truffle and ganache-cli)
- Removed matrix statement from travis.yml
- Removed the range for Pragma and changed to fixed 0.5.8 but it should be changed to 0.5.12
- Removed Unnecessary Getters from the CustomCrowdsale.sol Contract.
- Removed Unnecessary /** */ comments from the CustomCrowdsale.sol Contract.
- Fixed some issues regarding camel case in function naming.

AsiganTheSunk committed:
Remove unnecessary Getters

Remove unnecessary getters and properly propagate the changes to the Contract Test. Aditionally removed buyToken from the fallback methods

AsiganTheSunk committed:
Change except message on WETH9 test & Pragma Definition in Contracts

+ change except message weth9.js
+ changed the pragma version in solidity contracts.

AsiganTheSunk committed:
Remove matrix: from travis.yml

testing with unnecessary lines removed

AsiganTheSunk committed:
Modify travis.yml

AsiganTheSunk committed:
Update travis.yml

AsiganTheSunk committed:
Remove install script

remove install script install-travis.sh, moved the main install lines to the travis.yml file, test-contracts-travis.sh was not modified.

Uxio0 commented on .travis.yml:
```diff
@@ -2,24 +2,18 @@ dist: xenial
+  - os: linux
+    dist: xenial
+    before_install:
+      - rm -rf node_modules
```
Not needed

Uxio0 commented on client/src/contracts/ERC20Detailed.json:
```diff
@@ -247,31 +247,31 @@
     "absolutePath": "openzeppelin-solidity/contracts/token/ERC20/ERC20Detailed.sol",
```
Remove json contracts from git

Uxio0 commented on contracts/CustomCrowdsale.sol:
```diff
@@ -110,114 +101,57 @@ contract CustomCrowdsale is Ownable {
-
     // Total Tokens held in the CustomCrowdsale
     function getTokenTotalSupply() public view returns(uint) {
         return token.totalSupply();
```
You should call `balanceOf(address(this))`

Uxio0 commented on test/3_TestWETH9.js:
```diff
@@ -81,7 +81,7 @@ contract('WETH9', (accounts) => {
             await myWeth9Instance.deposit({'value': malicious_amount});
         } catch (err) {
-            console.log('       # ', err.message);
+            console.log('       (expect)  ', err.message);
```
Look at this example on how to test an event https://github.com/gnosis/safe-contracts/blob/0941fef007c2d9cb6a37e4b9fca5b7e65a486a44/test/gnosisSafeDeploymentViaCreate2.js#L56

Uxio0 reviewed (changes requested):
Good job overall!

AsiganTheSunk committed:
Fixed Requested Changes in fix_known_issues

+ Add .ignore path client/src/contracts 
+ Change getToken Balance in CustomCrowdsale.sol
+ Remove "rm -rf" since the instance is brand new every launch in travis

AsiganTheSunk committed:
Update client

+ Propagate change regarding function removal in contracts, set the autogenerated names for the getters in App.js
+ Moved Approval operation to the BuyToken section of the BuyForm.js to avoid having to store the cuantity of the current purchaes in a history

AsiganTheSunk committed:
Remove Contract Abis from client

Uxio0 commented on contracts/CustomCrowdsale.sol:
```diff
@@ -146,8 +146,8 @@ contract CustomCrowdsale is Ownable {
-    function getTokenTotalSupply() public view returns(uint) {
-        return token.totalSupply();
+    function getTokenTotalSupply(address src) public view returns(uint) {
+        return token.getBalance(src);
```
Why that src? If you want to get the balance for the token just call the token. I thought this method was used to check the tokens of the `CustomCrowdsale` contract

AsiganTheSunk committed:
Add Comments on Client

+ Add comment in App.js
+ Add comment in BuyForm.js and then a new button functionality. (you cannot submit if you dont put a valid nubmer in the box

AsiganTheSunk committed:
Update on Client

+ Add comment in App.js
+ Add comment in BuyForm.js and then a new button functionality. (you cannot submit if you dont put a valid nubmer in the box

AsiganTheSunk committed:
Revert "Merge remote-tracking branch 'origin/fix_known_issues' into fix_known_issues"

This reverts commit e8af9bfcaa0cca8a2770cf752ed3d4806c5942fc.

AsiganTheSunk committed:
Merge remote-tracking branch 'origin/fix_known_issues' into fix_known_issues

AsiganTheSunk committed:
Update Client & Test

* New * 
+ Add New Countdown for releaseTime
+ Add New Countdown for closeTime
+ Add New CrowdsaleInformation Point: TokensLeft
+ Add New CrowdsaleInformation Point: TokensLeftUntilGoal
+ Fix Typo in Type on CustomToken Test BN Expected, was not used.
+ Add new state variable to fill with Instance/Addr data of contracts
+ Add Ownership pooling to show a Close Crowdsale Button in BuyForm.js (Not Functioning).
+ Add behaviour in BuyForm.js the buttons will be disable until the proper amount is written in the input field. This does not apply to the Close Crowdsale Button.
+ Modify Disclaimer, enlarge width to better fit the current state of the Crowdsale landing page.
+ Add new process to cut the decimals from the view in the landing page, so it can be read more clearly.

* Pending *: 
+ modify and homogenize console.log messages
+ review and homogenize comments in the client

AsiganTheSunk committed:
Update package.json

AsiganTheSunk committed:
Fix Button Close Crowdsale

Now it properly shows only to the owner of the contract.

AsiganTheSunk committed:
Fix Typo in Crowdsale.sol

Fix typo in CustomCrowdsale.sol causing travis build to fail.

AsiganTheSunk committed:
Add New Restriction on BuyToken Function

Add new restriction in buyToken Function now should be equal or more than starting time

AsiganTheSunk committed:
Remove couner from the landing page

Remove counter from App.js, Component Crowdsale because sometimes It was a little bit misleading

AsiganTheSunk committed:
Add reload on account change

Add reload app.js on account change
